src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -1159,7 +1208,18 @@ static int do_map(Config *cfg)
     global_init_postfork_finish(g_ceph_context);
   }
 
+  watch_ctx = new WNBDWatchCtx(io_ctx, handler, image, info.size);
+  r = image.update_watch(watch_ctx, &handle);
+  if (r < 0)
+    goto close_ret;
```

Comment:
Wouldn't this jump straight to closing `image` and tearing down `rados` under a running `handler` and also potentially leave a stale registry entry behind?